driver/src/main/java/org/kaazing/k3po/driver/internal/control/handler/ControlServerHandler.java

```diff
@@ -229,6 +231,43 @@ public void abortReceived(final ChannelHandlerContext ctx, MessageEvent evt) thr
         robot.abort().addListener(whenAbortedOrFinished);
     }
 
+    @Override
+    public void notifyReceived(final ChannelHandlerContext ctx, MessageEvent evt) throws Exception {
+        NotifyMessage notifyMessage = (NotifyMessage) evt.getMessage();
+        final String barrier = notifyMessage.getBarrier();
+        if (logger.isDebugEnabled()) {
+            logger.debug("NOTIFY: " + barrier);
+        }
+        robot.notifyBarrier(barrier);
+        final NotifiedMessage notifiedMessaged = new NotifiedMessage();
+        notifiedMessaged.setBarrier(barrier);
+        logger.debug("sending NOTIFIED: " + barrier);
+        ChannelFuture pendingNotify = ctx.getChannel().write(notifiedMessaged);
+        pendingWrites.add(pendingNotify);
+    }
+
+    CopyOnWriteArrayList<ChannelFuture> pendingWrites = new CopyOnWriteArrayList<>();
```

Comment:
Oops, I don't think the pending writes are needed, it was supposed to be removed.  Thanks